tests/PhpCodeSnifferDrupalTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace OpenEuropa\CodeReview\Tests;
+
+use GrumPHP\Collection\FilesCollection;
+use GrumPHP\Task\Context\RunContext;
+
+/**
+ * Tests the PHP_CodeSniffer task using the Drupal conventions.
+ */
+class PhpCodeSnifferDrupalTest extends PhpCodeSnifferTestBase
+{
+    /**
+     * Tests Drupal code to make sure CodeSniffer triggers the appropriate errors.
+     *
+     * @param string $fixture
+     *   Name of the fixture.
+     * @param string $configuration
+     *   The name of the configuration to use in the task
+     * @param int    $expected
+     *   Expected result after the test.
+     *
+     * @dataProvider dataProvider
+     */
+    public function testDrupalPhpCodeSnifferDetector($fixture, $configuration, $expected)
+    {
+        $collection = new FilesCollection([$this->getFixture($fixture)]);
+        $context = new RunContext($collection);
+        $task = $this->getTask('phpcs', $configuration);
+        $result = $task->run($context);
+        $this->assertFailures($expected, $this->getFailures($result));
+    }
+
+    /**
+     * Test case provider function.
+     *
+     * Test file extensions.
+     *
+     * @return array
+     *      Test data.
+     */
+    public function dataProvider()
+    {
+        return [
+            [
+                'phpcs/DrupalClass.php',
+                'drupal-conventions',
+                [
+                    'error' => [
+                        18 => 1,
```

Comment:
18 is the line in code where errors were found. 1 is the amount of errors.

We should look for error messages, not lines and countings. Needs improvements.
I've even done some improvements but they need discussion and it would be out of scope so I've prevent myself from doing them.